cpp/src/arrow/buffer_builder.h

```diff
@@ -196,6 +202,14 @@ class ARROW_EXPORT BufferBuilder {
   int64_t length() const { return size_; }
   const uint8_t* data() const { return data_; }
   uint8_t* mutable_data() { return data_; }
+  template <typename T>
+  const T* data_as() const {
+    return reinterpret_cast<const T*>(data());
```

Comment:
Nit: just return `data_` so Debug builds don't have to put code to call `data()` :D